src/components/modal/z-modal-login/index.tsx

```diff
@@ -0,0 +1,233 @@
+import {
+  Component,
+  h,
+  Element,
+  Event,
+  EventEmitter,
+  Prop,
+} from "@stencil/core";
+import { HostElement, State } from "@stencil/core/internal";
+import {
+  ButtonVariantEnum,
+  InputStatusBean,
+  InputStatusEnum,
+} from "../../../beans";
+import { getSiblings } from "../../../utils/utils";
+
+/**
+ * @slot username - username input
+ * @slot password - password input
+ * @slot login - login button
+ * @slot signup - signup button
+ * @slot provider - expternal providers login buttons
+ */
+@Component({
+  tag: "z-modal-login",
+  styleUrl: "styles.css",
+  shadow: false,
+  scoped: true,
+})
+export class zModalLogin {
+  @Element() hostElement: HostElement;
+
+  /** Login modal title */
+  @Prop() heading?: string = "Entra in MyZanichelli";
+  /** Username/password input status */
+  @Prop({ mutable: true }) status?: InputStatusBean;
+  /** Username helper message */
+  @Prop({ mutable: true }) message?: string;
+
+  @State() externalProviderCheck: boolean = false;
+
+  componentDidLoad() {
+    this.externalProviderCheck = !!getSiblings(
+      // INFO: lo shadowDOM è aperto, si possono usare i selettori standard
+      this.hostElement.querySelector('slot-fb[name="provider"]')
+    ).filter((sibl) => sibl.nodeName !== "Z-BODY").length;
+  }
+
+  /** Emitted on login submit */
+  @Event() loginSubmit: EventEmitter;
+  emitLoginSubmit() {
+    const usernameInput = this.hostElement.querySelector(
+      "z-input#username"
+    ) as HTMLZInputElement;
+    const passwordInput = this.hostElement.querySelector(
+      "z-input#password"
+    ) as HTMLZInputElement;
+
+    // INFO: slotted inputs
+    if (!usernameInput || !passwordInput) return;
+
+    let username = usernameInput.value;
+    let password = passwordInput.value;
+
+    if (!username || !password) {
+      this.status = InputStatusEnum.error;
+      return;
+    }
+
+    username = this.cleanUsername(username);
+
+    this.loginSubmit.emit({
+      username,
+      password,
+      originalUsername: usernameInput.value,
+    });
+  }
+
+  /** Emitted on signup button click */
+  @Event() signupClick: EventEmitter;
+  emitSignupClick() {
+    this.signupClick.emit();
+  }
+
+  /** Emitted on zaino digitale button click */
+  @Event() zainoDigitaleClick: EventEmitter;
+  emitZainoDigitaleClick() {
+    this.zainoDigitaleClick.emit();
+  }
+
+  handleInputClick(e: KeyboardEvent) {
```

Comment:
The submit event is triggered only if you have a submit button inside the form (and, at the moment, we don't). Also, the submit events triggers a page reload, which is something that I believe this modal should not trigger.
I'll fix the method name, if it's ok with you.

Reply:
From the [docs](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event) I read this: 
> The submit event fires when the user clicks a submit button (`<button>` or `<input type="submit">`) or presses **Enter** while editing a field (e.g. `<input type="text">`) in a form.
I tried and it works with a simple `<input type="text">`.

To avoid the reload you can do `event.preventDefault()`. But nevermind, I just tried and for some reason it doesn't work here 😕

Just fix the method's name